### PR TITLE
[#57285] added shared examples for testing set permissions command

### PR DESCRIPTION
**Pull request by Kharonus**
# Ticket
[#57285](https://community.openproject.org/work_packages/57285)

# What are you trying to accomplish?
- having a consistent interface for both sharepoint and nextcloud command for setting permissions
- using a consistent test interface for the testable behaviour

# What approach did you choose and why?
- added shared examples for set permission commands
- removed deprecated authentication from onedrive util
- migrated one drive specs to new shared examples

### CODE REVIEW

File: modules/storages/spec/common/storages/peripherals/storage_interaction/one_drive/set_permissions_command_spec.rb
Comment (Kharonus) [inline modules/storages/spec/common/storages/peripherals/storage_interaction/one_drive/set_permissions_command_spec.rb:1]: this test now uses the shared examples. The previous VCR cassettes are still in use, as the overall requests didn't change.

File: modules/storages/spec/services/storages/project_storages/delete_service_spec.rb [hunk 1]
```diff
@@ -145,4 +139,14 @@
...
+
+  private
+
+  def command_class_reference
+    if storage.provider_type_nextcloud?
+      Storages::Peripherals::StorageInteraction::Nextcloud::DeleteFolderCommand
+    else
+      Storages::Peripherals::StorageInteraction::OneDrive::DeleteFolderCommand
+    end
+  end
```
Comment (mereghost) [inline modules/storages/spec/services/storages/project_storages/delete_service_spec.rb:151]: Can't we use the `Registry` to solve this here?

**Review by mereghost — APPROVED** [review]
Just a minor point. Looks great! =)

File: modules/storages/spec/services/storages/project_storages/delete_service_spec.rb (hunk 1 above)
Comment (Kharonus) [reply modules/storages/spec/services/storages/project_storages/delete_service_spec.rb:151]: well, honestly I'm not sure. I just fixed a rubocop here.

this is the code:

```ruby
    let(:command_double) { class_double(command_class_reference, call: ServiceResult.success) }

    before do
      Storages::Peripherals::Registry
        .stub("#{storage.short_provider_type}.commands.delete_folder", command_double)
    end
```

before it was an unverified double on `:delete_folder_command`. I understand we need the double to expect calls on it? Is this solvable with the registry alone?

Comment (Kharonus) [reply modules/storages/spec/services/storages/project_storages/delete_service_spec.rb:151]: Well, it works :D. It's just a bit weird, If I would inline it, then we would have this:

`Registry.stub("key", class_double(Registry.resolve("key"), call: Success))`

Is this weird? or fine? Asking for a friend.

Comment (mereghost) [reply modules/storages/spec/services/storages/project_storages/delete_service_spec.rb:151]: We only do `.call` over the command, right? So we could do:

```ruby
Registry.stub("key", ->(*) { Success })
```

lambdas respond to call and allow us full control. You could even set some expectations inside the lambda to validate arguments and what not.

**Pull request merged**